test/spotify-web-api.js

```diff
@@ -1249,6 +1249,88 @@ describe('Spotify Web API', function() {
       });
   });
 
+  it("should get user's devices:", function(done) {
+    sinon.stub(HttpManager, '_makeRequest', function(method, options, uri, callback) {
+      method.should.equal(superagent.get);
+      uri.should.equal('https://api.spotify.com/v1/me/player/devices');
+      options.headers.should.eql({Authorization: 'Bearer someAccessToken'});
+      callback(null, {
+        body : {
+          devices: [ ]
+        }
+      });
+    });
+
+    var api = new SpotifyWebApi({
+      accessToken : 'someAccessToken'
+    });
+
+    api.getMyDevices()
+      .then(function(data) {
+        should.exist(data.body.devices);
+        done();
+      });
+  });
+
+  it("should get user's current playback status:", function(done) {
+    sinon.stub(HttpManager, '_makeRequest', function(method, options, uri, callback) {
+      method.should.equal(superagent.get);
+      uri.should.equal('https://api.spotify.com/v1/me/player');
+      options.query.should.eql({
+        market : "GB"
+      });
+      options.headers.should.eql({Authorization: 'Bearer someAccessToken'});
+      callback(null, {
+        body : {
+          device: { }
+        }
+      });
+    });
+
+    var api = new SpotifyWebApi({
+      accessToken : 'someAccessToken'
+    });
+
+    api.getMyCurrentPlaybackState({ market : "GB"})
+      .then(function(data) {
+        should.exist(data.body.device);
+        done();
+      });
+  });
+
+  it('should transfer the user\'s playback', function(done) {
+
+    sinon.stub(HttpManager, '_makeRequest', function(method, options, uri, callback) {
+      method.should.equal(superagent.put);
+      uri.should.equal('https://api.spotify.com/v1/me/player');
+      JSON.parse(options.data).should.eql({
+        'device_ids': ['deviceId'],
+        'play': true,
+        'deviceIds' : ['deviceId']
```

Comment:
Why do both `deviceIds` and `device_ids` be `['deviceId']`?

Reply:
Ping @philnash :)

Reply:
The API [wants the parameter `device_ids`](https://developer.spotify.com/web-api/transfer-a-users-playback/) but in `_addBodyParameters` the rest of the optional parameters are added. I bet this can be fixed by deleting `deviceId` once it's added to the options, let me have a go at that.

Reply:
Cleaned up the request here: https://github.com/thelinmichael/spotify-web-api-node/pull/154